### PR TITLE
feat(secrets): age-encrypted *.age files in the apply pipeline

### PR DESCRIPTION
**Pull request by yukimemi**
Second half of the secret-management work — depends conceptually on but isn't blocked by #56 (mount-src absolute / Tera). Either lands first; merge ordering doesn't matter since the changes don't overlap.

## What

\`*.age\` ciphertext files in source are decrypted to a plaintext sibling (suffix dropped) on every \`yui apply\`, like \`*.tera\` rendering. The plaintext sibling lands in the same managed \`# >>> yui rendered <<<\` section of \`.gitignore\`; the apply walker links it to target like any other dotfile.

\`\`\`
home/.ssh/id_ed25519.age   <- committed ciphertext
     ↓ yui apply
home/.ssh/id_ed25519       <- decrypted sibling, gitignored
     ↓ link
~/.ssh/id_ed25519          <- target
\`\`\`

## Public surface

- **\`[secrets] identity\`** — path to your local age secret key file (default \`~/.config/yui/age.txt\`).
- **\`[secrets] recipients\`** — list of \`age1…\` public keys; feature is OFF when empty, so existing repos behave exactly as before this PR.
- **\`yui secret init [--comment TEXT]\`** — generate an X25519 keypair, write the secret with the same header format \`age-keygen\` produces, append the public key to \`config.local.toml\` \`[secrets].recipients\`. Idempotent.
- **\`yui secret encrypt <path> [--force] [--rm-plaintext]\`** — encrypt to all recipients. \`--rm-plaintext\` is gated on the plaintext living under \`\$DOTFILES\` so a typo can't erase files outside the repo.

## Drift detection

If the on-disk plaintext sibling diverges from the canonical \`.age\` (e.g. user edited the target which got auto-absorbed into the plaintext sibling), \`apply\` bails with a clear message. Fix: \`yui secret encrypt --force <path>\`. Mirrors the existing render-drift policy.

## Crypto + scope

**X25519 only in v1.** Plugin identities (YubiKey, FIDO2 / Pixel passkey, Touch ID, Windows Hello, TPM, 1Password) are real and useful but need callback plumbing for the plugin binaries' interactive prompts. Out of scope for this PR; the data model is plugin-ready so plugin support is additive.

## Test plan

- [x] 162 tests passing (8 secret module + 3 apply end-to-end + existing 151)
- [x] \`cargo make check\` (fmt + clippy \`-D warnings\` + locked check + tests)
- [x] Crypto round-trip with random keypair (single + multi recipient)
- [x] apply decrypt → sibling → link → target
- [x] apply bails on plaintext-vs-ciphertext drift
- [x] empty recipients list → secret walker is a no-op (existing repos unaffected)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added age-based encrypted secrets management with `yui secret init` to bootstrap and `yui secret encrypt` to secure files.
  * Secrets are automatically decrypted during `apply` with drift detection to ensure consistency.
  * Support for multi-machine recipient access.
  * Rendered plaintext secrets are automatically added to `.gitignore`.

* **Documentation**
  * Added comprehensive secrets documentation with setup and usage examples.

### CODE REVIEW

File: src/cmd.rs [hunk 1]
```diff
@@ -221,6 +238,25 @@ pub fn apply(source: Option<Utf8PathBuf>, dry_run: bool) -> Result<()> {
...
+            .iter()
+            .chain(render_report.unchanged.iter())
+            .chain(render_report.diverged.iter())
+            .chain(secret_report.managed_paths())
+            .cloned()
+            .collect();
+        managed.sort();
+        managed.dedup();
+        render::write_managed_section(&source, &managed)?;
+    }
```
[the remark this answers is not in the document]
Comment (yukimemi) [reply src/cmd.rs]: @gemini-code-assist fixed in 92ec39a. `render::render_all` no longer touches `.gitignore` as a side effect — callers (`cmd::apply`, `cmd::render`) drive a single deterministic write via `render::write_managed_section`. New `render::report_managed_paths` helper exposes the path set without coupling callers to RenderReport's field layout. Three render-module tests updated to invoke the helper explicitly.

File: src/cmd.rs [hunk 2]
```diff
@@ -627,6 +675,179 @@ pub fn unlink(source: Option<Utf8PathBuf>, paths_arg: Vec<Utf8PathBuf>) -> Resul
...
+            let after = &out[insertion..];
+            out = format!("{before}{comment_block}  \"{public}\",\n{after}");
+        } else {
+            // `[secrets]` exists but no `recipients = [...]` array yet.
+            out.push_str(comment_block);
+            out.push_str(&format!("recipients = [\"{public}\"]\n"));
+        }
+    }
+    out
+}
```
[the remark this answers is not in the document]
Comment (yukimemi) [reply src/cmd.rs]: @gemini-code-assist fixed in 92ec39a — switched to `toml_edit = "0.23"` (added as a direct dep). `append_recipient_to_local` now parses into a `DocumentMut`, inserts into `[secrets].recipients` correctly regardless of surrounding tables / comments / formatting, and serialises back preserving the user's layout. Four new unit tests cover the edge cases: secrets-table-missing, secrets-table-with-other-tables-around-it, secrets-table-without-recipients-array-yet, and idempotent re-add of the same key.
